### DISCUSSION TITLE
Creating a high quality voice on a CPU (no supported GPU)

### DISCUSSION THREAD

**Discussion by MrEdwards007** [coqui-ai/TTS#2180] [General Q&A]
I am looking to create the highest quality voice (identical sounding) possible.
I have a machine that does not have a supported GPU (MacBook), 16G of RAM, 2.3GHz i9.  

So, all training would have to be done on the CPU.
- **_Roughly_** how long should it take to go from ground zero to an identical sounding model?  
- How much training audio should I expect to provide (2, 5, 10, 20 hours of audio)? 
- What type of TTS output time should I expect, relative to real time?
- Once a model is created using a CPU, can it later be used against a GPU?
- Is there an API that you can run from Python or other language versus using the server, to automate outputs?

Thank you.

**Comment by king-dahmanus** [comment]
I'd love to eventually train a model on my low end laptop with 4gb of ram, an intel celeron 1.6ghz cpu lol. It will probably not work at all seeing my low ram and all

**Comment by MrEdwards007** [comment]
I've been going through the forums and it appears that this is not doable on a CPU.  However, in. order to understand what type of financial and time investment is needed, I'm hoping some of you have some rough ideas on the following.

I am looking to create the highest quality voice (identical sounding) possible.
If we assume an less expensive or average GPU...

-     Roughly how long should it take to go from ground zero to an identical sounding model?
-     How much training audio should I expect to provide (2, 5, 10, 20 hours of audio)?
-     What type of TTS output time should I expect, relative to real time?
-     Once a model is created using a CPU, can it later be used against a GPU?
-     Is there an API that you can run from Python or other language versus using the server, to automate outputs?


Thank you.
